### DISCUSSION TITLE
MS Graph Auth issues deployment GOV CLoud

### DISCUSSION THREAD

**Discussion by bfregy** [Azure/ipam#129] [Q&A]
I keep getting this below:

INFO: Granting admin consent for Microsoft Graph API permissions assigned to IPAM UI application
New-MgOauth2PermissionGrant_CreateExpanded1: C:\AzureDeploy\ipam-main\ipam-main\deploy\deploy.ps1:528:9
Line |
 528 |          New-MgOauth2PermissionGrant `
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Access token validation failure. Invalid audience.
INFO: Admin consent for Microsoft Graph API permissions granted successfully
INFO: Granting admin consent to the IPAM UI application for exposed API from the IPAM Engine application
New-MgOauth2PermissionGrant_CreateExpanded1: C:\AzureDeploy\ipam-main\ipam-main\deploy\deploy.ps1:544:7
Line |
 544 |        New-MgOauth2PermissionGrant `
     |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Access token validation failure. Invalid audience.
INFO: Admin consent for IPAM Engine exposed API granted successfully
INFO: Granting admin consent for Azure Service Management API permissions assigned to IPAM Engine application
New-MgOauth2PermissionGrant_CreateExpanded1: C:\AzureDeploy\ipam-main\ipam-main\deploy\deploy.ps1:562:7
Line |
 562 |        New-MgOauth2PermissionGrant `
     |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Access token validation failure. Invalid audience.
INFO: Admin consent for Azure Service Management API permissions granted successfully
INFO: Deploying IPAM bicep templates
ERROR: Unable to deploy Azure IPAM solution due to an exception, see logs for detailed information!

**Comment by DCMattyG** [comment]
Hey @bfregy, thank you so much for reaching out!

Just to clarify things a bit, you mentioned we're talking about Azure Gov. Is this IL5 or IL6?

Also, when connecting to Azure PowerShell, are you authenticating with your user from the Gov tenant, or from another account with guest access?

What does this command show?

```
(Get-AzContext).Environment.Name
```

**Comment by bfregy** [comment]
Hey there, thanks for the quick response. Deployment is in IL2 for a test for IL4/5/6 GOV

I am a global admin on all. Here's the info your requested

AzureUSGovernment

Authing w/GOV AAD Account  -- No sync of any on-prem accounts.

Lemme know if you need anything else?  BTW is MS Graph not going to graph.microsoft.us?




__________________________________
________________________________
From: DCMattyG ***@***.***>
Sent: Wednesday, May 31, 2023 19:12
To: Azure/ipam ***@***.***>

Subject: Re: [Azure/ipam] MS Graph Auth issues deployment GOV CLoud (Discussion #129)


Hey @bfregy<https://github.com/bfregy>, thank you so much for reaching out!

Just to clarify things a bit, you mentioned we're talking about Azure Gov. Is this IL5 or IL6?

Also, when connecting to Azure PowerShell, are you authenticating with your user from the Gov tenant, or from another account with guest access?

What does this command show?

(Get-AzContext).Environment.Name

**Reply by DCMattyG** [reply]
Indeed, if your Environment is showing "AzureUSGovernment", that does indeed map the Graph calls to graph.microsoft.us.

What version of Azure PowerShell are you currently using?

```
Get-InstalledModule -Name "Az"
```

**Reply by bfregy** [reply]
Version 10.0.0  -- but what bugs me is the " Access token validation failure. Invalid audience" in MS Graph -- Can I presume that is why the IPAM Bicep templates are crashing due to wrong creds ? or com v us??

**Reply by bfregy** [reply]
Also -- I did change the main.bicep param from AZURE_Cloud to AzureUSGovernment -- don't know if that was an issue.

**Reply by DCMattyG** [reply]
Honestly it's a bit hard to tell @bfregy and I'd love to work a bit more 1:1 with you to pinpoint what's going on here.

We absolutely have other customers running this in Azure Gov, so I'm curious as to why you're seeing this.

Would you mind dropping me an email at Matthew.Garrett@microsoft.com so we can find some time to connect?

I know we'll get you fixed up!
